compliance_checker/suite.py

```diff
@@ -511,87 +512,46 @@ def generate_dataset(self, cdl_path):
-        else:
-            def is_binary_string(bts):
-                # do a cheap imitation of libmagic
-                # http://stackoverflow.com/a/7392391/84732
-                if sys.version_info >= (3, ):
-                    join_str = ''
-                    textchars = join_str.join(map(chr, [7, 8, 9, 10, 12, 13, 27]
-                                                  + list(range(0x20, 0x100)))).encode()
-                else:
-                    # because of `unicode_literals` import, we need to convert
-                    # to a Py2 string/bytes
-                    join_str = str('')
-                    textchars = join_str.join(map(chr, [7, 8, 9, 10, 12, 13, 27]
-                                                  + list(range(0x20, 0x100))))
-                return bool(bts.translate(None, textchars))
-
-            def is_cdl_file(filename, data):
-                '''
-                Quick check for .cdl ascii file
-
-                Example:
-                    netcdf sample_file {
-                    dimensions:
-                        name_strlen = 7 ;
-                        time = 96 ;
-                    variables:
-                        float lat ;
-                            lat:units = "degrees_north" ;
-                            lat:standard_name = "latitude" ;
-                            lat:long_name = "station latitude" ;
-                    etc...
-
-                :param str filename: Absolute path of file to check
-                :param str data: First chuck of data from file to check
-                '''
-                if os.path.splitext(filename)[-1] == '.cdl':
-                    return True
-                if data.startswith(b'netcdf') or b'dimensions' in data:
-                    return True
-                return False
+        if pr.netloc:
+            return self.load_remote_dataset(ds_str)
+        return self.load_local_dataset(ds_str)
 
-            with open(ds_str, 'rb') as f:
-                first_chunk = f.read(1024)
-                if is_binary_string(first_chunk):
-                    # likely netcdf file
-                    pass
-                elif is_cdl_file(ds_str, first_chunk):
-                    ds_str = self.generate_dataset(ds_str)
-                else:
-                    f.seek(0)
-                    doc = f.read()
+    def load_remote_dataset(self, ds_str):
+        '''
+        Returns a dataset instance for the remote resource, either OPeNDAP or SOS
 
-        if doc is not None:
-            ds = self.process_doc(doc)
+        :param str ds_str: URL to the remote resource
+        '''
+        if opendap.is_opendap(ds_str):
+            return Dataset(ds_str)
         else:
-            # no doc? try the dataset constructor
-            ds = Dataset(ds_str)
+            response = requests.get(ds_str, allow_redirects=True)
+            if 'text/xml' in response.headers['content-type']:
+                return self.process_doc(response.content)
```

Comment:
I was wondering why checking for SOS didn't mirror the same `protocols` approach but then realized you want to operate on the response in one shot rather than request twice, that's fine.  Maybe a comment here how this is SOS just for discoverability in code.